pkg/trace/info/stats.go

```diff
@@ -155,15 +167,108 @@ func (ts *TagStats) publish() {
 	metrics.Count("datadog.trace_agent.receiver.events_extracted", eventsExtracted, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.events_sampled", eventsSampled, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.payload_accepted", requestsMade, tags, 1)
+
+	for reason, count := range ts.TracesDropped.tagValues() {
+		metrics.Count("datadog.trace_agent.normalizer.traces_dropped", count, append(tags, "reason:"+reason), 1)
+	}
+
+	for reason, count := range ts.TracesMalformed.tagValues() {
+		metrics.Count("datadog.trace_agent.normalizer.traces_malformed", count, append(tags, "reason:"+reason), 1)
+	}
+}
+
+// inlineNonZeroTagValues serializes the entries in this map into format "key1: value1, key2: value2, ...", sorted by
+// key to ensure consistent output order. Only non-zero values are included.
+func inlineNonZeroTagValues(statsMap map[string]int64) string {
+	keys := make([]string, len(statsMap))
+	i := 0
+	for k := range statsMap {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+
+	var results []string
+	for _, key := range keys {
+		value := statsMap[key]
+		if value > 0 {
+			results = append(results, fmt.Sprintf("%s:%d", key, value))
+		}
+	}
+
```

Comment:
Would you mind removing this empty line? I feel that the closure of the loop above it already creates enough spacing.

Reply:
Removed